prereise/gather/winddata/hrrr/grib.py

```diff
@@ -0,0 +1,111 @@
+    :param str variable: string designating what kind of data
+    :param str level: additional string that designates more information about
+        type of data
+    :param str forecast: designates how many hours into the future
+        this data is forecasted for
+    """
+
+    message_number: str
+    beginning_byte: str
+    ending_byte: str
+    initialization_date: str
+    variable: str
+    level: str
+    forecast: str
+
+    @classmethod
+    def from_string(cls, raw_string, next_raw_string=None):
+        """Creates a GribRecordInfo object from a specific string.
+        Information on the string format can be found at
+        `this link <https://github.com/blaylockbk/Herbie/blob/18945e4c5103386c98d08dcb2de590e2ac14c3d5/docs/user_guide/grib2.rst#how-grib-subsetting-works-in-herbie>`_
+
+        :param str raw_string: a record line following the format
+            MESSAGE_NUMBER:BEGINNING_BYTE:INITIALIZATION_DATE:VARIABLE:LEVEL:FORECAST
+        :param str next_raw_string: the following record line. This
+            is needed to know when the next record starts, so we know where
+            the current record ends.
+
+        :return: (*GribRecordInfo*)
+        """
+        details = raw_string.split(":")
+        assert len(details) >= 6
+        ending_byte = None
+        if next_raw_string:
+            ending_byte = str(int(next_raw_string.split(":")[1]) - 1)
+        return cls(
+            message_number=details[0],
+            beginning_byte=details[1],
+            ending_byte=ending_byte,
+            initialization_date=details[2],
+            variable=details[3],
+            level=details[4],
+            forecast=details[5],
+        )
+
+    @classmethod
+    def generate_grib_record_information_list(
+        cls, raw_record_information_list, index_list
+    ):
+        """Creates a list of GribRecordInfo objects from a
+        list of raw strings and a list of integers that designate
+        which raw strings to use to create GribRecordInfo objects.
+
+        :param list raw_record_information_list: a list of record lines that
+            follow the format:
+            MESSAGE_NUMBER:BEGINNING_BYTE:INITIALIZATION_DATE:VARIABLE:LEVEL:FORECAST
+        :param list index_list: a list of integers representing indices
+            to index into raw_record_information_list to create
+            GribRecordInfo objects out of
+
+        :return: (*list*) -- a list of GribRecordInfo objects
+        """
+        return_list = []
+        for index in index_list:
+            if index < len(raw_record_information_list) - 1:
+                return_list.append(
+                    cls.from_string(
+                        raw_record_information_list[index],
+                        raw_record_information_list[index + 1],
+                    )
+                )
+            else:
+                return_list.append(cls.from_string(raw_record_information_list[index]))
+        return return_list
```

Comment:
Maybe we can simply do following here:
```python
return [
    cls.from_string(raw_record_information_list[cur_ind], raw_record_information_list[next_ind])
    for cur_ind, next_ind in zip(raw_record_information_list, raw_record_information_list[1:])
]
```

Reply:
@BainanXia, it seems that you only want to consider a list of `GridRecordInfo` fo only a subset of `raw_record_information_list`

Reply:
> @BainanXia, it seems that you only want to consider a list of `GridRecordInfo` fo only a subset of `raw_record_information_list`

Yeah, nice catch, fixed. I assume there won't but index in `index_list` that is greater than the length of `raw_record_information_list`. Ignore it if that's not true.

Reply:
@goccert25 My previous code snippet above is wrong and pointed out by @rouille , now I think it should do the same thing as the original code and potentially be more cleaner.

Reply:
nice yeah